### PR TITLE
Fix the post LN bug

### PR DESCRIPTION
**Pull request by yidong72**
Fix a bug about how to do post-ln in the transformer layer. 
image

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/transformer.py [hunk 1]
```diff
@@ -1319,7 +1320,9 @@ def forward(
 
             layernorm_input = bias_dropout_add_func(attention_output, attention_bias, residual, self.hidden_dropout)
             normalization_output = self.post_inter_attention_layernorm(layernorm_input)
-
+            # Post-LN normalization after residual
+            if self.transformer_block_type == 'post_ln':
+                layernorm_input = normalization_output
```
Comment (michalivne) [inline nemo/collections/nlp/modules/common/megatron/transformer.py:1325]: Why not set directly `residual` here? why set `layernorm_input`?

**Review by michalivne — COMMENTED** [review]
See minor comment.

File: nemo/collections/nlp/modules/common/megatron/transformer.py (hunk 1 above)
Comment (yidong72) [reply nemo/collections/nlp/modules/common/megatron/transformer.py:1325]: because later logic set `residual = layernorm_input`

**Review by michalivne — APPROVED** [review]
LGTM!

**Review by ericharper — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**